File: packages/shared/src/today.ts
import { z } from "zod";

export const priorityItemOutSchema = z.object({
  id: z.string().uuid(),
  title: z.string(),
  priority_order: z.number().int(),
});

export const eventSummaryOutSchema = z.object({
  id: z.string().uuid(),
  item_id: z.string().uuid(),
  title: z.string(),
  start_at: z.string().datetime(),
  end_at: z.string().datetime(),
});

export const todayResponseSchema = z.object({
  priorities: z.array(priorityItemOutSchema),
  next_event: eventSummaryOutSchema.nullable(),
  next_action: priorityItemOutSchema.nullable(),
});

export type PriorityItemOut = z.infer<typeof priorityItemOutSchema>;
export type EventSummaryOut = z.infer<typeof eventSummaryOutSchema>;
export type TodayResponse = z.infer<typeof todayResponseSchema>;

File: apps/api/src/schemas/today.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PriorityItemOut(BaseModel):
    id: uuid.UUID
    title: str
    priority_order: int

    model_config = {"from_attributes": True}


class EventSummaryOut(BaseModel):
    id: uuid.UUID
    item_id: uuid.UUID
    title: str
    start_at: datetime
    end_at: datetime

    model_config = {"from_attributes": True}


class TodayResponse(BaseModel):
    priorities: list[PriorityItemOut]
    next_event: EventSummaryOut | None
    next_action: PriorityItemOut | None
